nixos/modules/services/hardware/spacenavd.nix

```diff
@@ -0,0 +1,26 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let cfg = config.hardware.spacenavd;
+
+in {
+
+  options = {
+    hardware.spacenavd = {
+      enable = mkEnableOption "spacenavd to support 3DConnexion devices";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.spacenavd = {
+      description = "spacenavd";
+      after = [ "syslog.target" ];
+      wantedBy = [ "graphical.target" ];
+      serviceConfig = {
+        ExecStart = "${pkgs.spacenavd}/bin/spacenavd -d -l syslog";
+        StandardError = "syslog";
```

Comment:
Oh and actually one importing thing I missed before: This service currently runs as root, which is a security concern. Can you try making `DynamicUser = true` work? That would be ideal

Reply:
This should be a a per-user service, which will make discovering the running X11 server etc. much easier (and the reason for why we patch it to use XDG_RUNTIME_DIR.